internal/provider/schemas/deployment.go

```diff
@@ -195,6 +195,13 @@ func DeploymentResourceSchemaAttributes() map[string]resourceSchema.Attribute {
 				stringplanmodifier.UseStateForUnknown(),
 			},
 		},
+		"desired_workload_identity": resourceSchema.StringAttribute{
+			MarkdownDescription: "Deployment's desired workload identity. The Terraform provider will use this provided workload identity to create the Deployment. If it is not provided the worload identity will be assigned automatically.",
+			Optional:            true,
+			PlanModifiers: []planmodifier.String{
+				stringplanmodifier.RequiresReplaceIfConfigured(),
```

Comment:
is this needed?

Reply:
not sure i'll test it without to see.

Reply:
good eyes. We do not want this as it causes the deployment to get deleted and recreated on change.

Reply:
I retested the update flow after removing this and all looks good.